### PR TITLE
Implement aws.ecs.* resource attributes

### PR DESCRIPTION
**Pull request by mmanciop**
# Description

Implement the [experimental AWS ECS resource attributes](https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/cloud_provider/aws/ecs/) using the Metadata v4 Endpoint.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [X] Unit tests

General logic on Amazon ECS has been extensively tested by Lumigo in their [OpenTelemetry Python Distro](https://github.com/lumigo-io/opentelemetry-js-distro), which has shipped this detection logic for some time already.

# Does This PR Require a Core Repo Change?

- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [X] Changelogs have been updated
- [X] Unit tests have been added
- [ ] Documentation has been updated <-- There seems to be no need of that

### CODE REVIEW

**Comment by mmanciop** [comment]
There's discussion on https://github.com/open-telemetry/opentelemetry-java/pull/4574#issuecomment-1213233736 whether the ARNs like the one this PR builds are actually correct. Until we are use, let's not merge :D

**Comment by mmanciop** [comment]
**Update:** The uncertainty about ARNs seems sorted out

**Comment by Kausik-A** [comment]
Running the ECS UNIT tests ( test_ecs.py ) seems to fail for `test_ecs.AwsEcsResourceDetectorTest`. This is what I get `AttributeError: <module 'opentelemetry.sdk.extension.aws.resource.ecs' from '/usr/local/lib/python3.9/site-packages/opentelemetry/sdk/extension/aws/resource/ecs.py'> does not have the attribute '_http_get'`
Can you run them again locally and lmk if we have to follow any specified steps for the UNIT tests to pass.

**Comment by mmanciop** [comment]
> Running the ECS UNIT tests ( `test_ecs.py` ) seems to fail for `test_ecs.AwsEcsResourceDetectorTest`. This is what I get `AttributeError: <module 'opentelemetry.sdk.extension.aws.resource.ecs' from '/usr/local/lib/python3.9/site-packages/opentelemetry/sdk/extension/aws/resource/ecs.py'> does not have the attribute '_http_get'` Can you run them again locally and lmk if we have to follow any specified steps for the UNIT tests to pass.

@Kausik-A, I am running the tests locally with `tox -e py39-test-sdkextension-aws`, which is how I believe the CI runs them. When running `pytest`, there is a whole bunch of entirely-unrelated tests that fail, so I am guessing plain pytest is not really supported.

**Comment by mmanciop** [comment]
> As done in the [opentelemetry-go-contrib](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/2626#issuecomment-1216710348) can we add we add test cases that utilize the [fargate examples responses](https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-metadata-endpoint-v4-fargate.html)?

Done in d945a61

**Comment by mmanciop** [comment]
@Kausik-A I squashed, rebased and run the tests again; care to have a look?

**Comment by mmanciop** [comment]
@Kausik-A fixed the linter warnings

**Comment by srikanthccv** [comment]
@Kausik-A would you be willing to add yourself to CODEOWNERS for this package since @NathanielRN hasn't been very active here?

**Comment by Kausik-A** [comment]
> @Kausik-A fixed the linter warnings and rebased due to changelog conflicts

@mmanciop thank you for the update.Let me have another pass at it

**Comment by Kausik-A** [comment]
> @Kausik-A would you be willing to add yourself to CODEOWNERS for this package since @NathanielRN hasn't been very active here?

@srikanthccv are you referring to being codeowners for [opentelemetry-sdk-extension-aws](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/2de26c81750401dd85d02f1ebe4c3501ae2a967c/sdk-extension)?

**Comment by srikanthccv** [comment]
Yes, then your review counts into merge requirements.

**Comment by Kausik-A** [comment]
Yeah, I can do that

**Comment by mmanciop** [comment]
@Kausik-A have a look please?

**Comment by Kausik-A** [comment]
LGTM! @srikanthccv can you take a look at this [permission-request-PR ](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1379) so that I can approve this?

**Comment by Kausik-A** [comment]
@srikanthccv can you approve the running workflow for this PR? I still can't seem to add my review.

**Comment by srikanthccv** [comment]
You can leave your approval review on the changes.

**Comment by mmanciop** [comment]
@Kausik-A I fixed hopefully the last linting issue

**Comment by ocelotl** [comment]
@mmanciop please fix the conflicts :v:

**Comment by mmanciop** [comment]
@ocelotl done

**Comment by mmanciop** [comment]
@ocelotl @NathanielRN @Aneurysm9

File: sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/resource/ecs.py
Comment (Aneurysm9) [inline sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/resource/ecs.py:144]:
```suggestion
                    logger.warning("Cannot parse AWS region out of ECS ARN")
```

**Review by Aneurysm9 — APPROVED** [review]

File: CHANGELOG.md
Comment (mmanciop) [reply CHANGELOG.md]: Done in 1ae794dec2a9589114d0f6dbe8b160932c868846

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**
